src/global.rs

```diff
@@ -36,6 +48,44 @@ extern "C" {
 pub struct Global<T> {
   value: Option<NonNull<T>>,
   isolate_handle: Option<IsolateHandle>,
+  weakable: Option<Rc<dyn Weakable<T>>>,
+}
+
+pub type WeakCallback<T> = extern "C" fn(NonNull<T>, NonNull<Isolate>);
+
+/// Used to retrieve the callback and argument for weakened `Global` pointers.
+pub unsafe trait Weakable<T> {
```

Comment:
Can you document the invariants that have to be upheld to implement this unsafe trait? I.e. why does it have to be unsafe?